cli/src/main.rs

```diff
@@ -184,7 +189,22 @@ fn main() {
     }
 }
 
-fn get_file_paths(input_patterns: Vec<String>) -> Result<Vec<PathBuf>, GlobError> {
+fn get_file_paths(input_patterns: Vec<String>, exclude_patterns: Vec<String>) -> Result<Vec<PathBuf>, GlobError> {
+    let exclude_patterns = exclude_patterns
+        .into_iter()
+        .filter_map(|exclude_pattern| {
+            let is_dir = fs::metadata(&exclude_pattern)
+                .map(|meta| meta.is_dir())
+                .unwrap_or(false);
+            let global_pattern = if is_dir {
+                format!("{}/**/*", &exclude_pattern.trim_end_matches('/'))
+            } else {
+                exclude_pattern
+            };
+            Pattern::new(&global_pattern).ok()
+        })
+        .collect::<Vec<_>>();
```

Comment:
how about extracting the duplicate code in a function, like so:

```suggestion
    fn as_glob_pattern(pattern: String) -> String {
        let is_dir = fs::metadata(&pattern)
            .map(|meta| meta.is_dir())
            .unwrap_or(false);

        if is_dir {
            format!("{}/**/*", &pattern.trim_end_matches('/'))
        } else {
            pattern
        }
    };

    let exclude_patterns = exclude_patterns
        .into_iter()
        .map(as_glob_pattern)
        .map(|p| Pattern::new(&p))
        .collect::<Result<Vec<_>, _>>()
        .expect("failed to parse exclude glob pattern");
```

and reuse it for input patterns:
```rust
    input_patterns
        .into_iter()
        .map(as_glob_pattern)
        .flat_map(|glob_pattern| {
            // ...
  ```

Reply:
better :+1: